lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2704,10 +2742,150 @@ SILFunction *PartitionCloner::lookupSendReceiveFunction(StringRef fnName,
   return fn;
 }
 
+/// The partitioned code is illustrated by the snippet below:
+///
+/// Host side:
+/// bb0:
+///   ...
+///   swich_enum %X, case ..: bb1, case ..: bb2, .., case ..: bbN
+///
+/// bb1:
+///   sendToAccelerator(0) // 0 is the case id here
+///   <other host code>
+///   br bb_merge
+///
+/// <same for bb2 through bbN, sending a different case_id each>
+///
+/// bb_merge:
+///   ...
+///
+/// Accelerator side:
+/// bb0:
+///   ...
+///   %case_id = recvFromHost()
+///   if %case_id == 0 goto bb1
+///   else if %case_id == 1 goto bb2
+///   else ...
+///
+void PartitionCloner::handleSwitchEnum(SwitchEnumInst *inst) {
+  auto &ctx = FP.hostFn.getASTContext();
+  auto intDecl = ctx.getInt64Decl();
+  auto intFieldSILType = getStdlibNumericTypeFromDecl(intDecl);
+
+  auto hostLoc = getUserSourceLocation(inst);
+  auto *createScalarTensorFn =
+      lookupSendReceiveFunction("scalar", intFieldSILType, hostLoc);
+  // If `result` were not a send'able data type like String, compiler
+  // would have rejected the program before reaching here.
+  assert(createScalarTensorFn &&
+         "At this point of compilation, scalar value must be send'able "
+         "from Swift to TensorFlow.");
+  (void)createScalarTensorFn;
+
+  SILFunction *sendFn =
+      lookupSendReceiveFunction("sendToAccelerator", intFieldSILType, hostLoc);
+  assert(sendFn && "At this point of compilation, the value must be send'able "
+                   "from Swift to TensorFlow.");
+  (void)sendFn;
```

Comment:
`sendFn` is being used in the for-loop below. It seems unnecessary to try to silence 'unused' warnings here.

Reply:
Good catch. Removed.